### PR TITLE
Release 0.4.0-incubating

### PR DESCRIPTION
**Pull request by cfmcgrady**

### CODE REVIEW

File: docs/community/release_notes/release_note_0.4.0.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+  - navigation
+
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
```
Comment (pan3793) [inline docs/community/release_notes/release_note_0.4.0.md]: a blank line before and after it

File: docs/community/release_notes/release_note_0.4.0.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+  limitations under the License.
+---
+
+# Apache Celeborn(Incubating) 0.4.0 Release Notes
+
+## Highlight
+
+- Rerun Spark Stage for Celeborn Shuffle Fetch Failure
+- Added support for Hadoop MapReduce
+- Added support for Flink 1.18
```
Comment (pan3793) [inline docs/community/release_notes/release_note_0.4.0.md]: let's respect the Apache trademark
```suggestion
- Added support for Apache Hadoop MapReduce
- Added support for Apache Flink 1.18
```
Also, the site needs to update
image

File: docs/community/release_notes/release_note_0.4.0.md [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+- [CELEBORN-1134] Celeborn Flink client should validate whether execution.batch-shuffle-mode is ALL_EXCHANGES_BLOCKING
+- [CELEBORN-1106] Ensure data is written into flush buffer before sending message to client
+- [CELEBORN-1110][FOLLOWUP] Support celeborn.worker.storage.disk.reserve.ratio to configure worker reserved ratio for each disk
+- [CELEBORN-1108][FOLLOWUP] Use rat plugin check Flink 1.18
+- [CELEBORN-1048][FOLLOWUP] MR module compile
+- [CELEBORN-1108] Rat plugin check for more modules
+- [CELEBORN-247][FOLLOWUP] Add metrics for each user's quota usage of Celeborn Worker
+- [CELEBORN-1095] Support configuration of fastest available XXHashFactory instance for checksum of Lz4Decompressor
+- [CELEBORN-1087] Remove SimpleStateMachineStorageUtil in master module
+- Revert "[CELEBORN-255] Add counter of outstandingFetches, outstanding…
```
Comment (pan3793) [inline docs/community/release_notes/release_note_0.4.0.md]: such revert should not be present in release notes ...

let's remove it and the reverted one

File: docs/community/release_notes/release_note_0.4.0.md [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+- [CELEBORN-843][BUILD] `sbt` support flink-related module build/test
+- [CELEBORN-867][BUILD] Add maven local repository to sbt respositories
+- [CELEBORN-836][BUILD] Initial support sbt
+
+### Documentation
+
+- [CELEBORN-1207] SBT http repository documentation
+- [CELEBORN-1223] Align master and worker metrics of document with MasterSource and WorkerSource
+- [CELEBORN-1247] Output config's alternatives to doc
+- Add the Celeborn Helm Chart doc
```
Comment (pan3793) [inline docs/community/release_notes/release_note_0.4.0.md]: no ticket?

Comment (cfmcgrady) [reply docs/community/release_notes/release_note_0.4.0.md]: the [PR-2149](https://github.com/apache/incubator-celeborn/pull/2149) was merged by @FMX

File: docs/community/release_notes/release_note_0.4.0.md (hunk 3 above)
Comment (cfmcgrady) [reply docs/community/release_notes/release_note_0.4.0.md]: removed

File: docs/community/release_notes/release_note_0.4.0.md (hunk 4 above)
Comment (pan3793) [reply docs/community/release_notes/release_note_0.4.0.md]: for such cases, let's add a [MINOR] prefix

File: docs/community/release_notes/release_note_0.4.0.md [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+- [CELEBORN-822][DOC] Add quick start guide
+
+### Dependencies
+
+- [CELEBORN-1184] Update the snakeyaml version from 1.33 to 2.2
+- [CELEBORN-1173] Upgrade netty version from 4.1.93.Final to 4.1.101.Final
+- [CELEBORN-1163] Upgrade protobuf from 3.19.2 to 3.21.7
+- [CELEBORN-1169] Bump Spark from 3.4.1 to 3.4.2
+- [CELEBORN-1170] Upgrade snappy-java from 1.1.8.2 to 1.1.10.5
+- upgrade snappy-java from 1.1.8.2 to 1.1.10.5
```
Comment (pan3793) [inline docs/community/release_notes/release_note_0.4.0.md]: ditto

File: .github/workflows/site.yaml [hunk 6]
```diff
@@ -57,6 +57,8 @@ jobs:
           ./.github/bin/build_docs.sh 'tags/v0.2.1-incubating.tar.gz' '0.2.1-incubating'
       - run: |
           ./.github/bin/build_docs.sh 'tags/v0.3.2-incubating.tar.gz' '0.3.2-incubating'
+      - run: |
+          ./.github/bin/build_docs.sh 'tags/v0.4.0-incubating.tar.gz' '0.4.0-incubating'
```
Comment (pan3793) [inline .github/workflows/site.yaml]: inline it and above steps
```suggestion
      - run: ./.github/bin/build_docs.sh 'tags/v0.4.0-incubating.tar.gz' '0.4.0-incubating'
```

**Review by pan3793 — APPROVED** [review]
Almost good, but let other members take a good, especially for the Highlight part

**Review by cxzl25 — APPROVED** [review]

File: docs/community/release_notes/release_note_0.4.0.md (hunk 2 above)
Comment (cfmcgrady) [reply docs/community/release_notes/release_note_0.4.0.md]: would you help to update the website code? @waitinfuture

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**